Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py

```diff
@@ -0,0 +1,408 @@
+        mails_fetched: A list of Email objects
+        messages: A list of the ids of the messages fetched
+    """
+    if message_id:
+        messages = [message_id]
+    else:
+        messages_query = generate_search_query(first_fetch_time,
+                                               permitted_from_addresses,
+                                               permitted_from_domains)
+        messages = client.search(messages_query)
+        limit = len(messages) if limit == -1 else limit
+        messages = messages[:limit]
+    mails_fetched = []
+    for mail_id, message_data in client.fetch(messages, 'RFC822').items():
+        message_bytes = message_data.get(b'RFC822')
+        if not message_bytes:
+            continue
+        # The search query filters emails by day, not by exact date
+        email_message_object = Email(message_bytes, include_raw_body, save_file, mail_id)
+        if not first_fetch_time or email_message_object.date > first_fetch_time:
+            mails_fetched.append(email_message_object)
+    return mails_fetched, messages
+
+
+def generate_search_query(latest_created_time: Optional[datetime],
+                          permitted_from_addresses: str,
+                          permitted_from_domains: str) -> list:
+    """
+    Generates a search query for the IMAP client 'search' method. with the permitted domains, email addresses and the
+    starting date from which mail should be fetched.
+    Input example:
+    latest_created_time: datetime.datetime(2020, 8, 7, 12, 14, 32, 918634, tzinfo=datetime.timezone.utc)
+    permitted_from_addresses: ['test1@mail.com', 'test2@mail.com']
+    permitted_from_domains: ['test1.com', 'domain2.com']
+    output example:
+    ['OR',
+     'OR',
+     'OR',
+     'FROM',
+     'test1@mail.com',
+     'FROM',
+     'test2@mail.com',
+     'FROM',
+     'test1.com',
+     'FROM',
+     'domain2.com',
+     'SINCE',
+     datetime.datetime(2020, 8, 7, 12, 14, 32, 918634, tzinfo=datetime.timezone.utc)]
+    Args:
+        latest_created_time: The greatest incident created_time we fetched from last fetch
+        permitted_from_addresses: A string representation of list of mail addresses to fetch from
+        permitted_from_domains: A string representation list of domains to fetch from
+
+    Returns:
+        A list with arguments for the email search query
+    """
+    permitted_from_addresses_list = argToList(permitted_from_addresses)
+    permitted_from_domains_list = argToList(permitted_from_domains)
+    messages_query = ''
+    if permitted_from_addresses_list + permitted_from_domains_list:
+        messages_query = OR(from_=permitted_from_addresses_list + permitted_from_domains_list).format()
+        # Removing Parenthesis and quotes
+        messages_query = messages_query.strip('()').replace('"', '')
+    # Creating a list of the OR query words
+    messages_query_list = messages_query.split() + ['SINCE', latest_created_time]  # type: ignore[list-item]
```

Comment:
can we convert `latest_created_time` to str and remove the type ignore ?

Reply:
No, it must be a datetime object